src/main/java/io/jenkins/plugins/pipelinegraphview/cards/RunDetailsItem.java

```diff
@@ -2,85 +2,83 @@
 
 import static java.util.Objects.requireNonNull;
 
-public class RunDetailsItem {
-
-    private final String icon;
-    private final String text;
-    private final String href;
-    private final boolean separator;
-    private final String tooltip;
-
-    RunDetailsItem(String icon, String text, String href, boolean separator, String tooltip) {
-        this.icon = icon;
-        this.text = text;
-        this.href = href;
-        this.separator = separator;
-        this.tooltip = tooltip;
-    }
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
 
-    public String getIcon() {
-        return icon;
-    }
+public sealed interface RunDetailsItem {
 
-    public String getText() {
-        return text;
-    }
+    RunDetailsItem SEPARATOR = new Separator();
 
-    public String getHref() {
-        return href;
-    }
+    record Separator() implements RunDetailsItem {}
 
-    public String getTooltip() {
-        return tooltip;
-    }
+    final class RunDetail implements RunDetailsItem {
+        private final @NonNull Icon icon;
+        private final @NonNull ItemContent content;
+        private final @Nullable String tooltip;
 
-    public boolean isSeparator() {
-        return separator;
-    }
+        public RunDetail(@NonNull Icon icon, @NonNull ItemContent content, @Nullable String tooltip) {
+            this.icon = requireNonNull(icon);
+            this.content = requireNonNull(content);
+            this.tooltip = tooltip;
+        }
 
-    public static class Builder {
+        public RunDetail(@NonNull Icon icon, @NonNull ItemContent content) {
+            this(icon, content, null);
+        }
 
-        private String icon;
-        private String text;
-        private String href;
-        private String tooltip;
-        private boolean separator;
+        @Nullable
+        public String tooltip() {
+            return tooltip;
+        }
 
-        public Builder text(String text) {
-            this.text = text;
-            return this;
+        @NonNull
+        public ItemContent content() {
+            return content;
         }
 
-        public Builder icon(String icon) {
-            this.icon = icon;
-            return this;
+        @NonNull
+        public String icon() {
+            return icon.value();
         }
+    }
 
-        public Builder tooltip(String tooltip) {
-            this.tooltip = tooltip;
-            return this;
+    sealed interface Icon {
+        String value();
+
+        record SimpleIcon(@NonNull String value) implements Icon {
+            public SimpleIcon {
+                requireNonNull(value);
+            }
         }
 
-        public Builder ionicon(String ionicon) {
-            this.icon = String.format("symbol-%s plugin-ionicons-api", ionicon);
-            return this;
+        record IonIcon(@NonNull String value) implements Icon {
```

Comment:
this is a library and it should be named after it properly: https://ionic.io/ionicons (possibly with an s after not sure)
```suggestion
        record Ionicon(@NonNull String value) implements Icon {
```